crates/bevy_pbr/src/meshlet/from_mesh.rs

```diff
@@ -194,53 +196,69 @@ fn compute_meshlets(indices: &[u32], vertices: &VertexDataAdapter) -> Meshlets {
     meshlets
 }
 
-fn collect_triangle_edges_per_meshlet(
+fn find_connected_meshlets(
     simplification_queue: Range<usize>,
     meshlets: &Meshlets,
-) -> HashMap<usize, HashSet<(u32, u32)>> {
-    let mut triangle_edges_per_meshlet = HashMap::new();
-    for meshlet_id in simplification_queue {
+) -> HashMap<usize, Vec<(usize, usize)>> {
+    // for each edge, gather all meshlets that use it
+    let mut edges_to_meshlets = HashMap::new();
+
+    for meshlet_id in simplification_queue.clone() {
         let meshlet = meshlets.get(meshlet_id);
-        let meshlet_triangle_edges = triangle_edges_per_meshlet
-            .entry(meshlet_id)
-            .or_insert(HashSet::new());
         for i in meshlet.triangles.chunks(3) {
-            let v0 = meshlet.vertices[i[0] as usize];
-            let v1 = meshlet.vertices[i[1] as usize];
-            let v2 = meshlet.vertices[i[2] as usize];
-            meshlet_triangle_edges.insert((v0.min(v1), v0.max(v1)));
-            meshlet_triangle_edges.insert((v0.min(v2), v0.max(v2)));
-            meshlet_triangle_edges.insert((v1.min(v2), v1.max(v2)));
+            for k in 0..3 {
+                let v0 = meshlet.vertices[i[k] as usize];
+                let v1 = meshlet.vertices[i[(k + 1) % 3] as usize];
+                let edge = (v0.min(v1), v0.max(v1));
+
+                let vec = edges_to_meshlets
+                    .entry(edge)
+                    .or_insert_with(SmallVec::<[usize; 2]>::new);
+                // meshlets are added in order, so we can just check the last element to deduplicate
+                if vec.last() != Some(&meshlet_id) {
```

Comment:
This took me a minute to figure out. Can we change the end of this comment to something like the following? "last element to deduplicate, in the case of two triangles sharing the same edge within a single meshlet"

Reply:
Added verbatim, thanks